src/Stratis.Bitcoin/P2P/PeerConnectorAddNode.cs

```diff
@@ -52,20 +52,20 @@ public PeerConnectorAddNode(NodeSettings nodeSettings, IPeerAddressManager peerA
         /// <inheritdoc/>
         public override NetworkAddress FindPeerToConnectTo()
         {
-            PeerAddress peer = null;
-
             foreach (var endPoint in this.NodeSettings.ConnectionManager.AddNode)
             {
-                peer = this.peerAddressManager.FindPeer(endPoint.MapToIpv6());
+                PeerAddress peer = this.peerAddressManager.FindPeer(endPoint.MapToIpv6());
                 if (peer == null)
                     continue;
 
                 // If the peer is already connected just continue.
-                if (this.IsPeerConnected(endPoint))
+                if (this.IsPeerConnected(endPoint.MapToIpv6()))
```

Comment:
Instead of having 2 `continue;` here, if there's a way to get the value of `endPoint.MapToIpv6()` from `peer`, it would be clearer to have a single statement such as 
```
if(peer == null || this.IsPeerConnected(peer.NetworkAddress))
    continue;

return peer.NetworkAddress;
```

Reply:
Can't you just do `if(peer == null || this.IsPeerConnected(endPoint.MapToIpv6()))` in that case?
But to me this looks fine.

If any refactoring I would suggest to put all inside if statements 
```
if(peer != null)
  if([connected])
     return peer
```

Reply:
See the new code changes ;)